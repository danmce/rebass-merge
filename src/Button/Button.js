import React from 'react'
import PropTypes from 'prop-types'
import { Button as RebassButton } from 'rebass'

const Button = ({ children, stretch, ...props }) => (
  <RebassButton width={stretch ? '100%' : 'auto'} {...props}>
    {children}
  </RebassButton>
)

export default Button

Button.propTypes = {
  onClick: PropTypes.func,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]),
  variant: PropTypes.oneOf(['primary', 'secondary', 'outline']),
  stretch: PropTypes.bool
}

Button.defaultProps = {
  variant: 'primary'
}
