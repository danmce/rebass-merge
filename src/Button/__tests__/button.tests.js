import React from 'react'
import * as enzyme from 'enzyme'
import toJson from 'enzyme-to-json'
import Button from '../Button'

test('renders Button correctly', () => {
  const button = <Button>Test</Button>

  expect(toJson(enzyme.mount(button))).toMatchSnapshot('enzyme.mount')
})
