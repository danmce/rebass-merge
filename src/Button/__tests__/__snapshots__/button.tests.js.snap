// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders error Button correctly: enzyme.mount 1`] = `
<Button
  variant="primary"
>
  <ForwardRef
    variant="primary"
    width="auto"
  >
    <Styled(div)
      __css={
        Object {
          "appearance": "none",
          "bg": "primary",
          "border": 0,
          "borderRadius": 4,
          "color": "white",
          "display": "inline-block",
          "fontSize": "inherit",
          "lineHeight": "inherit",
          "px": 3,
          "py": 2,
          "textAlign": "center",
          "textDecoration": "none",
        }
      }
      as="button"
      tx="buttons"
      variant="primary"
      width="auto"
    >
      <button
        className="emotion-0"
      >
        Test
      </button>
    </Styled(div)>
  </ForwardRef>
</Button>
`;
